### PR TITLE
Hwcodec: codec preference

### PR DESCRIPTION
**Pull request by 21pages**
1. codec preference
2. linux compatible
3. Code optimization
* check hwcodec config every time server or client start, but it is not checked when the tray icon started the service.
* Simple function 'best' in hwcodec, always check hwcodec in another process.

### CODE REVIEW

File: src/ui/header.tis [hunk 1]
```diff
@@ -157,6 +160,13 @@ class Header: Reactor.Component {
...
                 <li #low type="image-quality"><span>{svg_checkmark}</span>{translate('Optimize reaction time')}</li> 
                 <li #custom type="image-quality"><span>{svg_checkmark}</span>{translate('Custom')}</li>
+                {show_codec ? <div>
+                <div .separator />
+                <li #auto type="codec-preference"><span>{svg_checkmark}</span>Auto</li>
+                <li #vp9 type="codec-preference"><span>{svg_checkmark}</span>VP9</li>
+                {codecs[0] ? <li #h264 type="codec-preference"><span>{svg_checkmark}</span>H264</li> : ""}
+                {codecs[1] ? <li #h265 type="codec-preference"><span>{svg_checkmark}</span>H265</li> : ""}
+                </div> : ""}
                 <div .separator />
```
Comment (rustdesk) [inline src/ui/header.tis:170]: test

File: libs/hbb_common/protos/message.proto [hunk 2]
```diff
@@ -434,9 +440,17 @@ enum ImageQuality {
...
 message VideoCodecState {
-  int32 ScoreVpx = 1;
-  int32 ScoreH264 = 2;
-  int32 ScoreH265 = 3;
+  enum PerferCodec {
+    Auto = 0;
+    VPX = 1;
+    H264 = 2;
+    H265 = 3;
+  }
```
Comment (rustdesk) [inline libs/hbb_common/protos/message.proto:448]: test again

**Review by rustdesk — CHANGES_REQUESTED** [review]

File: src/ui/header.tis (hunk 1 above)
Comment (21pages) [reply src/ui/header.tis:170]: Do not display any options when only VP9 is available?

Comment (rustdesk) [reply src/ui/header.tis:170]: Correct

Comment (21pages) [reply src/ui/header.tis:170]: using the installed version, a CMD window will flash by at the controlled end when connected or switching the codec. This needs to be modified.

**Pull request merged**